platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/hash/VirtualHasher.java

```diff
@@ -400,7 +460,7 @@ public Hash hash(
                     while (curStackPath < Math.min(curPath, lastPathInCurStackChunk)) {
                         final ChunkHashTask t = map.remove(curStackPath);
                         assert t != null;
-                        t.setData(null);
+                        t.complete();
                         curStackPath++;
                     }
                     stack[curRank] = INVALID_PATH;
```

Comment:
As I wrote in #11636 (see item 3 in the bug description), this may be improved, if curPath is less than lastPathInCurStackChunk (that is, the current path is in the same chunk as the path in the stack), stack[curRank] may be set to curPath + 1